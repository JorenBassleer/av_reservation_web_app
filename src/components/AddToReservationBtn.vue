<template>
    <div>
        <button 
            class="add-btn" 
            :class="{animate: disabled}" 
            @click.prevent="addToReservation(appliance)"
        >
        Add
    </button>
    </div>
</template>
<script setup>
import { ref, defineProps } from 'vue';
import { useStore } from 'vuex';
const props = defineProps({
	appliance: {
		type: Object,
		required: true,
	}
});
const store = useStore();
const disabled = ref(false);
const addToReservation = (appliance) => {
    disabled.value = true;
    setTimeout(() => {
        disabled.value = false;
    }, 1000);
    store.commit('pushApplianceInReservation', appliance);
}
</script>
<style scoped>
.add-btn {
    position: absolute;
    margin-top: -60px;
    margin-left: 175px;
    z-index: 9999999999;
    cursor: pointer;
    font: inherit;
    border: none;
    outline: inherit;
    color: #e1dcce;
    border-radius: 8px;
    padding: 10px;
    background: #473f2b;
}
.animate {
    animation: expand 0.3s cubic-bezier(0.5,0.3,0.4,0.8) both;
}
@keyframes expand {
    50% {
        transform: scale(1.25);
    }

}
</style>