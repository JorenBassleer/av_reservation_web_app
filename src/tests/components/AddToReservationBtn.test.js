import { createStore } from 'vuex';
import { mount } from '@vue/test-utils';
import {
  expect, describe, it, beforeEach, vi,
} from 'vitest';
import AddToReservationBtn from '../../components/AddToReservationBtn.vue';

describe('AddToReservationBtn.vue', () => {
  // Arrange
  let wrapper;
  let mockStore;
  let mockedIncFn;
  const initReservation = {
    appliances: [],
  };
  const initAppliance = {
    id: '1561',
    name: 'test-appliance',
    brand: {
      name: 'test-brand',
    },
    type: {
      name: 'test-type',
    },
  };
  beforeEach(() => {
    vi.resetAllMocks();
    mockedIncFn = vi.fn();
    mockStore = createStore({
      state: { initReservation },
      mutations: { pushApplianceInReservation: mockedIncFn },
    });
  });
  it('calls pushApplianceInReservation store mutation', async () => {
    wrapper = mount(AddToReservationBtn, {
      global: { plugins: [mockStore] },
      props: { appliance: initAppliance },
    });
    // Act
    wrapper.find('button').trigger('click');
    expect(mockedIncFn).toHaveBeenCalled();
  });
  it('addToReservation func gets called', () => {
    wrapper = mount(AddToReservationBtn, {
      global: { plugins: [mockStore] },
      props: { appliance: initAppliance },
    });
    const spy = vi.spyOn(wrapper.vm, 'addToReservation');
    // Act
    wrapper.find('button').trigger('click');
    // Assert
    expect(spy).toHaveBeenCalled();
  });
});
