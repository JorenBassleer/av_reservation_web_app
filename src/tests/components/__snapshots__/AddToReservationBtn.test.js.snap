// Vitest Snapshot v1

exports[`AddToReservationBtn.vue > Match snapshot 1`] = `"<div data-v-9348ee48=\\"\\"><button class=\\"add-btn animate\\" data-v-9348ee48=\\"\\"> Add </button></div>"`;
